view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java

```diff
@@ -215,29 +211,44 @@ public static Schema convertAvroSchemaToPinotSchema(
         fieldSpecs.add(fieldSpec);
       } else if (timeColumn.equals(field.name())) {
         validateDateTimeColumn(field, timeUnit);
+        DateTimeFormatSpec dateTimeFormatSpec = DateTimeFormatSpec.forEpoch(timeUnit.name());
+        String dataTimeFormatStr =
+            String.format(
+                "%s:%s:%s",
+                dateTimeFormatSpec.getColumnSize(),
+                dateTimeFormatSpec.getColumnDateTimeTransformUnit().name(),
+                dateTimeFormatSpec.getTimeFormat().name());
+        String granularityStr =
+            String.format(
+                "%s:%s",
+                dateTimeFormatSpec.getColumnSize(),
+                dateTimeFormatSpec.getColumnDateTimeTransformUnit().name());
         fieldSpec =
             new DateTimeFieldSpec(
                 field.name(),
                 AvroUtils.extractFieldDataType(field),
-                new DateTimeFormatSpec(1, timeUnit.name(), TimeFormat.EPOCH.name()).getFormat(),
-                new DateTimeGranularitySpec(1, timeUnit).getGranularity());
+                dataTimeFormatStr,
```

Comment:
no longer accepts a format spec, and the format spec doesn't know how to convert itself to a string?

Reply:
Yes, thats my understanding. The toString representation and required format in input for DateTimeFieldSpec is different.